test/WithQueries.test.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import { render, waitForElement, cleanup } from 'react-testing-library';
+import { queryStrict, refetch, invalidate } from '../src/DSL';
+import { taskEither } from 'fp-ts/lib/TaskEither';
+import { WithQueries } from '../src/react';
+import { QueryResult } from '../src/QueryResult';
+
+describe('declareQueries', () => {
+  it('should work', async () => {
+    const foo = queryStrict(() => taskEither.of<void, string>('foo'), refetch);
+    const Foo = () => (
+      <WithQueries
+        queries={{ foo }}
+        render={queries =>
+          queries.fold(
+            () => 'loading',
+            () => 'failure',
+            ({ foo }) => foo
+          )
+        }
+      />
+    );
+
+    const { getByText } = await render(<Foo />);
+    await waitForElement(() => getByText('foo'));
+    cleanup();
+  });
+
+  it('should rerender after an invalidate', async () => {
+    const res = { value: 'foo' };
+    const foof = jest.fn(() => taskEither.of<void, string>(res.value));
+    const foo = queryStrict(foof, refetch);
+    const renderf = jest.fn(
+      (queries: QueryResult<void, { foo: typeof foo._P }>) =>
+        queries.fold(
+          () => 'loading',
+          () => 'failure',
+          ({ foo }) => foo
+        )
+    );
+    const Foo = () => <WithQueries queries={{ foo }} render={renderf} />;
+
+    const element = React.createElement(Foo);
+    const { getByText, rerender } = await render(element);
+    await waitForElement(() => getByText('foo'));
+    // why 3 and not 2?
+    // Currently WithQueries (implemented using declareQueries) subscribes in componentDidMount, after
+    // the first render which has already happened using monoidResult.empty as query result.
+    // Since we a) don't have a way of comparing result (i.e. no resultSetoid) and b) it's unsafe
+    // to call subscribe and potentially trigger a setState before the component is mounted,
+    // this is expected
+    expect(renderf.mock.calls.length).toBe(3);
```

Comment:
In case you missed it, there's a specific matcher for this in Jest:

```suggestion
    expect(renderf).toHaveBeenCalledTimes(3);
```